c/lincoa_c.f90

```diff
@@ -80,10 +84,22 @@ subroutine lincoa_c(cobj_ptr, data_ptr, n, x, f, cstrv, m_ineq, Aineq, bineq, m_
 bineq_loc = real(bineq, kind(bineq_loc))
 Aeq_loc = real(transpose(Aeq), kind(Aeq_loc))
 beq_loc = real(beq, kind(beq_loc))
-xl_loc = real(xl, kind(xl_loc))
-xu_loc = real(xu, kind(xu_loc))
-rhobeg_loc = real(rhobeg, kind(rhobeg_loc))
-rhoend_loc = real(rhoend, kind(rhoend_loc))
+if (C_ASSOCIATED(xl)) then
+    call C_F_POINTER(xl, xl_loc_interm, shape=[n])
+    call safealloc(xl_loc, int(n, IK))
+    xl_loc = real(xl_loc_interm, kind(xl_loc))
+end if
+if (C_ASSOCIATED(xu)) then
+    call C_F_POINTER(xu, xu_loc_interm, shape=[n])
+    call safealloc(xu_loc, int(n, IK))
+    xu_loc = real(xu_loc_interm, kind(xu_loc))
+end if
+if (.not. is_nan(rhobeg)) then
```

Comment:
Similar to what was mentioned before, I do not think it is necessary to check whether `rhobeg` or `rhoend` is NaN. Thank you for taking a look.